src/CloudEventSdkCompliant.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+use BadMethodCallException;
+use JsonSerializable;
+use CloudEvents\V1\CloudEventInterface;
+use DateTimeImmutable;
+use DateTimeInterface;
+
+/**
+ * @internal
+ * Wraps a Google\CloudFunctions\CloudEvent to comply with
+ * CloudEvents\V1\CloudEventInterface.
+ */
+class CloudEventSdkCompliant implements JsonSerializable, CloudEventInterface
```

Comment:
nit: newline spacing is inconsistent with these functions

Reply:
I'm not sure how this is going to be used.

Do we want to implement `CloudEventImmutable` in addition to `CloudEventInterface`?

Reply:
I think it's the getters don't have newlines in between them, but other functions do.

`CloudEventInterface` is going to be what's passed to CloudEvent Functions with the new declarative function signatures. This is similar to how `ServerRequestInterface` is passed to HTTP functions today.

`CloudEventImmutable` is not an interface, so we can't implement it. I don't think we would benefit from inheriting or using it within this class, since we are not using any other part of the CloudEvents SDK today that can serialize/deserialize to `CloudEvent` or `CloudEventImmutable`.

Using the `CloudEventInterface` will put us a step in the right direction, because it means we can swap our implementation to use the official CloudEvent SDK in the future, without rewiring everything all at once.

Reply:
Here's the next PR queue'd up which makes use of this new class, you can see the new function stub in this test:
https://github.com/anniefu/functions-framework-php/pull/1/files#diff-479b688739465688c5b60552045cb322a656e9bd47a2b9b12a05f8c4f76d3bd6R74